### DISCUSSION TITLE
Breakdown Profile and PermissionSet Metadata Into Subdirectories

### DISCUSSION THREAD

**Discussion by pgumiela** [forcedotcom/cli#2376] [Ideas]
**Is your feature request related to a problem? Please describe.**
Maintaining Profiles and Permission Sets is really problematic. We can't selectively set what should be included and what's not. If there is any problem with order of lines in these files, we need to waste a lot of time reviewing this back and forth. **It would be much easier to split this down, so we have separate subdirectories for every type of permission.**

**Related Salesforce Idea:**
https://trailblazer.salesforce.com/ideaView?id=0873A0000003acNQAQ

**What are you trying to do**
We want to make it easier to keep Profiles and Permission Sets in the code repo. Objects are already broken down into subdirectories, and we just need to apply the same idea for Profiles and Permission Sets.

**Describe the solution you'd like**
Starting out with smaller .xml files broken down into subdirectories is a step in the right direction. For some of these subdirectories, it could be beneficial to be able to manage this on the file level. For instance, if you want to add a new field to the Profile with Read-Only access, you have your `Profile/fieldPermissions/FooBar.field-meta.xml`, and it just contains Read and Edit properties.

**Describe alternatives you've considered**
It's really hard to consider any reasonable alternative. Managing such large .xml files is just not convenient.

**Additional Context**
People have proposed similar stuff already:
https://github.com/lodossDev/sfdx-profiles-splitter

[a comment by an automated account is not shown]

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-9860066](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000004aLlOYAU/view)

**Comment by thvd** [comment]
Not even mentioning dealing with merge conflicts in Profiles or Permissionsets

**Comment by jefersonchaves** [comment]
Is something to improve the comparison of Permission Sets and/or Profiles in the next year or so?

**Comment by mshanemc** [comment]
related discussion: https://github.com/forcedotcom/cli/discussions/2544

**Comment by rdietrick** [comment]
I've written an sfdx plugin that can break down profiles into more granular files and then put them back together before deployment.  
https://www.npmjs.com/package/@rdietrick/sfdx-profile-decompose
https://github.com/rdietrick/sfdx-profile-decompose
